src/expr.rs

```diff
@@ -1909,6 +1909,60 @@ pub mod parsing {
         }
     }
 
+    macro_rules! impl_trivial_parse {
```

Comment:
I don't think trivial is a useful way to describe these impls. I would recommend something like `impl_by_parsing_expr` for the macro name.